src/org/traccar/protocol/SviasProtocolDecoder.java

```diff
@@ -0,0 +1,150 @@
+
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.helper.PatternBuilder;
+
+import java.net.SocketAddress;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.regex.Pattern;
+import org.traccar.DeviceSession;
+import org.traccar.helper.Parser;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.Position;
+
+public class SviasProtocolDecoder extends BaseProtocolDecoder {
+
+    private static final String MSG_KEEPALIVE = "@";
+
+    public SviasProtocolDecoder(SviasProtocol protocol) {
+        super(protocol);
+    }
+
+    private static final Pattern PATTERN = new PatternBuilder()
+            .text("[")       // delimiter init
+            .number("(dddd),") // version hardware
+            .number("(dddd),") // version software
+            .number("(d+),") // counter
+            .number("(d+),") // imei
+            .any()            // model or hourmeter
+            .number("(d+),") // date (yyyymmdd)
+            .number("(d+),") // time (hhmmss)
+            .number("(-?d+),") // longitude
+            .number("(-?d+),") // latitude
+            .number("(d+),") // speed
+            .number("(d+),") // course
+            .number("(d+),") // odometer
+            .number("(d+),") // input
+            .number("(d+),") // output / status
+            .number("(d+),") // flag pack input 1
+            .number("(d+),") // flag pack input 2
+            .number("(d+),") // main power voltage
+            .number("(d+),") // percentual power internal battery
+            .number("(d+),") // RSSID
+            .any()
+            .compile();
+
+    private double convertCoordinates(long v) {
+        return Double.valueOf(((float) ((((float) v / 1.0E7F)
+                - ((int) (v / 10000000L))) * 1.6666666666666667D)) + ((int) (v / 10000000L)));
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg)
+            throws Exception {
+
+        String sentence = (String) msg;
+
+        if (!sentence.contains(":")) {
+
+            Parser parser = new Parser(PATTERN, (String) sentence);
+            if (!parser.matches()) {
+                return null;
+            }
+
+            Position position = new Position(getProtocolName());
+
+            String versionHard = parser.next();
+            String versionSoft = parser.next();
+            String counterInternal = parser.next();
+
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
+            if (deviceSession == null) {
+                return null;
+            }
+
+            position.setDeviceId(deviceSession.getDeviceId());
+
+            DateFormat dateFormat = new SimpleDateFormat("ddMMyyHHmmss");
+            dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+            String date = String.format("%06d", parser.nextInt());

```

Comment:
There is a better way to parse time directly with parser. Please see other decoders for examples.

Reply:
The date and time are integers, without leading zero.

Please give examples of how to proceed!

Reply:
There are plenty of examples in other decoders. If leading zeroes are missing you can easily use zero as default value in the parser.